### ISSUE TITLE
grpc support

### ISSUE THREAD

**Issue by szuecs** [zalando/skipper#906]
Maybe one day we could add grpc support
https://blogs.dropbox.com/tech/2019/01/courier-dropbox-migration-to-grpc/

**Comment by jkatzhrs** [comment]
Could this be expanded to generic protocol frontend/backend support? Essentially allow writing of shims that take in and emit requests?

**Comment by szuecs** [comment]
@jkatzhrs  yes, but I don't have the time to do this that generic. I think it would involve a big refactoring.
If you like to work on that I would be happy to review and test it.
I don't think it is easy for a first time contributor, but if you like give it a try.

**Comment by mariusv** [comment]
Using [ArgoCD](https://argoproj.github.io/argo-cd) without GRPC support is quite complicated as the only option I see so far for the CLI is to use port forwarding and then point the ArgoCD CLI to it which makes things complicated. ArgoCD requirements for ingress are either [GRPC or SSL-Passthrough](https://argoproj.github.io/argo-cd/operator-manual/ingress/). 
Any plans to support this soon? Or do you have any other suggestions ? Thanks

**Comment by szuecs** [comment]
I don’t know your environment, but if you’re using Kubernetes, I would suggest to service type loadbalancer.
If GRPC can work through http upgrade similar to spdy or h2, it would be possible if you enable upgrade headers support which is marked experimental but I am right now confident that it is good enough.
